Enhance flashblock processing metrics and caching logic

jowparks opened the pull request:
- Introduced caching for the maximum index of flashblocks per block.
- Updated metrics to track the number of flashblocks created and the maximum index per block.
- Added unit tests to verify the correct storage of maximum indices and metric recording for flashblocks across multiple blocks.

jonschlinkert commented:
> I know you don't like so much big examples,

I do it sometimes too, like you've said, sometimes it's necessary. I just like them a bit smaller in general. this is great though thanks!

tunnckoCore commented:
> damn, good catch.

Haha, yea! ;d

Btw, i tried to run verb, but it eats the whole readme (meaning.. it was like copy of verb.md file, with no other template staff from default layout from readme generator and etc), so i left it off.

jonschlinkert commented:
> it was like copy of verb.md file, with no other template staff from default layout from readme generator and etc), so i left it off.

yeah I have fixes for verb that will probably resolve that. the good thing is that verb will be really easy to maintain. the bugfixes are getting easier

tunnckoCore commented:
> the good thing is that verb will be really easy to maintain. the bugfixes are getting easier

absolutely! when we get familiar with underling libs/plugins. initially it cost some time (low) to realize from where comes given problem, but it worths.

jowparks committed:
Enhance flashblock processing metrics and caching logic

- Introduced caching for the maximum index of flashblocks per block.
- Updated metrics to track the number of flashblocks created and the maximum index per block.
- Added unit tests to verify the correct storage of maximum indices and metric recording for flashblocks across multiple blocks.
